tests/UtilTests/CanonicalTests.elm

```diff
@@ -0,0 +1,175 @@
+module UtilTests.CanonicalTests exposing (all)
+
+import AWS.Http exposing (..)
+import AWS.Signers.Canonical exposing (..)
+import Expect
+import Json.Encode as JE
+import Test exposing (describe, test, Test)
+import TestHelpers exposing (expectMatches)
+
+
+all : Test
+all =
+    describe "Canonical"
+        [ canonicalQueryStringTests
+        , canonicalUriTests
+        , canonicalHeadersTests
+        , signedHeadersTests
+        , canonicalPayloadTests
+        , canonicalTests
+        ]
+
+
+canonicalTests : Test
+canonicalTests =
+    describe "canonical"
+        [ test "does the same request encoding as http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html" <|
+            \_ ->
+                (canonical "get"
+                    ""
+                    [ ( "Content-Type", "application/x-www-form-urlencoded;  charset=utf-8" )
+                    , ( "x-amz-date", "20150830T123600Z" )
+                    , ( "host", "  iam.amazonaws.com" )
+                    ]
+                    (QueryParams
+                        [ ( "Version", "2010-05-08" )
+                        , ( "Action", "ListUsers" )
+                        ]
+                    )
+                )
+                    |> Expect.equal """f536975d06c0309214f805bb90ccff089219ecd68b2577efef23edd43b7e1a59"""
```

Comment:
Canonical request hasher is now well tested. I discovered and fixed some bugs will unit testing the module. But I have confidence in the implementation now because this test passes. This is the example provided in the AWS docs for creating a canonical request: http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html